### PR TITLE
product get functionality added

### PR DESCRIPTION
**Pull request by sauravces**
get all functionality added  with controller ,service ,repository layer . with tdd approach.

### CODE REVIEW

File: ProductPOC/Controllers/ProductController.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        public async Task<ActionResult<IEnumerable<ProductDto>>> GetProducts()
+        {
+            var products = await _productService.GetAllProductsAsync();
+            var productDtos = products.Select(p => new ProductDto
+            {
+                Id = p.Id,
+                Name = p.Name,
+                Description = p.Description,
+                Price = p.Price
+            });
```
Comment (aashikvilla) [inline ProductPOC/Controllers/ProductController.cs]: We can use auto mapper or even better to use mongo projection

File: ProductPOC/DbContext/ProductDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        private readonly IMongoDatabase _database;
+        public ProductDbContext(IConfiguration configuration)
+        {
+            var connectionString = configuration.GetConnectionString("MongoDb");
+           var client = new MongoClient(connectionString);
+           _database = client.GetDatabase(configuration["DatabaseSettings:DatabaseName"]);
+        }
+        
+
+        public IMongoCollection<Product> Products => _database.GetCollection<Product>("Products");
```
Comment (aashikvilla) [inline ProductPOC/DbContext/ProductDbContext.cs]: I think better to store collection name as well in appsettings

File: ProductPOC/appsettings.json [hunk 3]
```diff
@@ -5,5 +5,12 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings": {
+    "MongoDb": "mongodb+srv://sauravbalasahebchoudhari:37JoCLVY7OSMX3dc@cluster0.lyiyfwj.mongodb.net/"
```
Comment (aashikvilla) [inline ProductPOC/appsettings.json]: You can use user-secrets, avoid committing it to git

File: ProductPOC/bin/Debug/net8.0/AWSSDK.Core.dll
Comment (aashikvilla) [inline ProductPOC/bin/Debug/net8.0/AWSSDK.Core.dll:1]: Can you remove all these build files, probably use git-ignore

File: ProductTest/ProductTestRepository.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+            // Create an instance of ProductRepository with the mocked ProductDbContext
+            var repository = new ProductRepository(mockContext.Object);
+
+            // Act
+            var result = await repository.GetAllAsync();
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeEquivalentTo(products);
```
Comment (aashikvilla) [inline ProductTest/ProductTestRepository.cs]: Feels like not null assert id redundant, when are anyway checking for value to be equivalent to products

File: ProductPOC/DbContext/ProductDbContext.cs (hunk 2 above)
Comment (sauravces) [reply ProductPOC/DbContext/ProductDbContext.cs]: made this changes.

File: ProductPOC/Controllers/ProductController.cs (hunk 1 above)
Comment (sauravces) [reply ProductPOC/Controllers/ProductController.cs]: updated code with auto mapper.

File: ProductTest/ProductTestRepository.cs (hunk 4 above)
Comment (sauravces) [reply ProductTest/ProductTestRepository.cs]: ok sure.

File: ProductPOC/appsettings.json (hunk 3 above)
Comment (sauravces) [reply ProductPOC/appsettings.json]: updated code with user-secrets.

File: ProductPOC/bin/Debug/net8.0/AWSSDK.Core.dll
Comment (sauravces) [reply ProductPOC/bin/Debug/net8.0/AWSSDK.Core.dll:1]: ok sure.

File: ProductTest/bin/Debug/net8.0/AWSSDK.Core.dll
Comment (kool7) [inline ProductTest/bin/Debug/net8.0/AWSSDK.Core.dll:1]: Can we put these files in gitignore?

**Review by kool7 — CHANGES_REQUESTED** [review]

File: ProductTest/bin/Debug/net8.0/AWSSDK.Core.dll
Comment (sauravces) [reply ProductTest/bin/Debug/net8.0/AWSSDK.Core.dll:1]: yes

File: ProductPOC/appsettings.json (hunk 3 above)
Comment (aashikvilla) [reply ProductPOC/appsettings.json]: I can still see the connection string , are these changes in another PR?

Comment (sauravces) [reply ProductPOC/appsettings.json]: yes check latest PR 3 in which i updated.

Comment (aashikvilla) [reply ProductPOC/appsettings.json]: Ideally you are supposed to resolve in this PR so that we can merge this.

Comment (aashikvilla) [reply ProductPOC/appsettings.json]: But fine, you can merge rest PRs into this branch, so we would have one PR with all fixes

Comment (sauravces) [reply ProductPOC/appsettings.json]: yes,  PR3 is having code of PR1 and PR2 also.

**Pull request merged**